backend/src/main/java/ai/verta/modeldb/project/FutureProjectDAO.java

```diff
@@ -0,0 +1,149 @@
+package ai.verta.modeldb.project;
+
+import ai.verta.common.KeyValue;
+import ai.verta.common.ModelDBResourceEnum;
+import ai.verta.modeldb.AddExperimentRunTags;
+import ai.verta.modeldb.DeleteExperimentRunAttributes;
+import ai.verta.modeldb.DeleteExperimentRunTags;
+import ai.verta.modeldb.GetAttributes;
+import ai.verta.modeldb.GetTags;
+import ai.verta.modeldb.LogAttributes;
+import ai.verta.modeldb.common.connections.UAC;
+import ai.verta.modeldb.common.futures.FutureGrpc;
+import ai.verta.modeldb.common.futures.FutureJdbi;
+import ai.verta.modeldb.common.futures.InternalFuture;
+import ai.verta.modeldb.exceptions.PermissionDeniedException;
+import ai.verta.modeldb.experimentRun.subtypes.AttributeHandler;
+import ai.verta.modeldb.experimentRun.subtypes.TagsHandler;
+import ai.verta.uac.Action;
+import ai.verta.uac.IsSelfAllowed;
+import ai.verta.uac.ModelDBActionEnum;
+import ai.verta.uac.ResourceType;
+import ai.verta.uac.Resources;
+import ai.verta.uac.ServiceEnum;
+import java.util.Calendar;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.Executor;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class FutureProjectDAO {
+  private static Logger LOGGER = LogManager.getLogger(FutureProjectDAO.class);
+
+  private final Executor executor;
+  private final FutureJdbi jdbi;
+  private final UAC uac;
+
+  private final AttributeHandler attributeHandler;
+  private final TagsHandler tagsHandler;
+
+  public FutureProjectDAO(Executor executor, FutureJdbi jdbi, UAC uac) {
+    this.executor = executor;
+    this.jdbi = jdbi;
+    this.uac = uac;
+
+    String entityName = "ProjectEntity";
+    attributeHandler = new AttributeHandler(executor, jdbi, entityName);
+    tagsHandler = new TagsHandler(executor, jdbi, entityName);
+  }
+
+  public InternalFuture<Void> deleteAttributes(DeleteExperimentRunAttributes request) {
```

Comment:
Bad copy and paste

Reply:
Yes, agree it is resolved in each endpoints individual PR https://github.com/VertaAI/modeldb/pull/2118

Reply:
fixed